### ISSUE TITLE
Shortcuts on macOS?

### ISSUE THREAD

**Issue by Derekmini** [Dominic-DallOsto/zotero-reading-list#9]
Hello, may I ask if shortcut keys can be used on macOS?

I tried using Command+Num, Option+Num, Ctrl+Num but there was no response.

**Comment by Dominic-DallOsto** [comment]
I had a quick look and the Option key should work. Does [this](https://stackoverflow.com/a/52799323) code snippet work for you?

**Comment by Derekmini** [comment]
I press `Option + N` to obtain the desired result as:

![](https://oss-cn-beijing-derek.oss-cn-beijing.aliyuncs.com/img/202305091412669.png)

I pressed `Option + Num` again in Zotero, but it only highlighted the collection of items instead of setting the reading status.

**Comment by Dominic-DallOsto** [comment]
That's interesting - it seems like an existing shortcut is taking 
precedence and stopping us from detecting the shortcut.

I checked here (https://www.zotero.org/support/kb/keyboard_shortcuts) 
but didn't see this combination - only that Option+Num should change the 
annotation tool in the PDF Reader.
Is it possible another addon is implementing the shortcut to highlight a 
collection when you press Option+Num?

**Comment by Derekmini** [comment]
I have disabled all addons except for reading-list, and checked if other background software is using this keyboard shortcut combination. However, I still cannot add Read Status by pressing Option+Num.

![](https://oss-cn-beijing-derek.oss-cn-beijing.aliyuncs.com/img/202305091945036.png)

I feel that my previous description was unclear. Actually, it is the Option key that triggers the highlight. Pressing the Num key interrupts the continuous highlight. Here is a recording demonstrating this situation:

![](https://oss-cn-beijing-derek.oss-cn-beijing.aliyuncs.com/img/202305092033182.gif)

**Comment by ItsMeVictor** [comment]
Same problem here. Thank you for the add-on though! It's great. It'd be even better once the shortcuts work :)-)

**Comment by Dominic-DallOsto** [comment]
Thanks for the patience!

Are you able to test the latest version for Zotero 7 (#17) - does the issue still occur?

**Comment by itsmevictor** [comment]
@Dominic-DallOsto I have not yet switched to Zotero 7 as I am waiting for all the add-ons I'm using to be working on the new version, but I will let you know as soon as I have made the switch!

**Comment by jamy3531** [comment]
> Thanks for the patience!
> 
> Are you able to test the latest version for Zotero 7 (#17) - does the issue still occur?

This problem still occurs. I'm now using the latest version Zotero 7 and the latest plugin.

**Comment by Dominic-DallOsto** [comment]
Can you paste this into your browser console (F12 dev tools) and see if you can get the key presses when using Option+Num

```javascript
window.addEventListener('keydown', function(e) {
  if (e.altKey && !e.ctrlKey && !e.shiftKey && ["0","1","2","3","4","5","6","7","8","9"].includes(e.key))
    console.log(e.key); 
});
```

If that works and you see numbers being logged, can you paste the same code in Tools > Developer > Run Javascript in Zotero. Then you do you see numbers in Tools > Developer > Error Console when you press that key combination?

**Comment by adamslc** [comment]
Hi there, sorry for the duplicate issue. I wasn't able to get that code to work in Zotero, but I was able to use the Firefox dev console. I held Alt, and then pressed 1 through 0, and got the following results:
```
key: Alt, alt: true, ctrl: false, shift: false
key: ¡, alt: true, ctrl: false, shift: false
key: ™, alt: true, ctrl: false, shift: false
key: £, alt: true, ctrl: false, shift: false
key: ¢, alt: true, ctrl: false, shift: false
key: ∞, alt: true, ctrl: false, shift: false
key: §, alt: true, ctrl: false, shift: false
key: ¶, alt: true, ctrl: false, shift: false
key: •, alt: true, ctrl: false, shift: false
key: ª, alt: true, ctrl: false, shift: false 
key: º, alt: true, ctrl: false, shift: false
```

It looks like the issue is that MacOS is doing a transformation of the key before it is received by the application. Would it be possible to make the keyboard shortcuts configurable?

**Comment by Dominic-DallOsto** [comment]
I found a similar issue [here](https://stackoverflow.com/questions/14109639/how-to-get-alt-cor-any-other-key-to-work-in-javascript-on-chrome-for-mac-and#14109721) which states that pressing Opt + key on Mac also modifies the keycode instead of returning the original keycode.

**Comment by Dominic-DallOsto** [comment]
This has been fixed in v1.3.2 of this plugin, but the fix seems a bit "fragile" / like it might be keyboard / language dependent.

Please let me know if you have any issues in your configuration.
